### PR TITLE
Module 10

### PR DESCRIPTION
**Pull request by waormd**
Module 10 work

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,42 +1,110 @@
...
     app = Flask(__name__)
 
     @app.route('/', methods=['GET'])
+    @login_required
     def indexGet():
-        item_view_model = ViewModel(mongo_api.load_todo_items())
-        return render_template('index.html', view_model = item_view_model)
+        current_user_role = check_user()
+        if (current_user_role == 'writer' or current_user_role == 'reader'):
+            item_view_model = ViewModel(mongo_api.load_todo_items(), current_user_role)
```
Comment (hugh-emerson) [inline app.py:39]: This check seems unnecessary when there's no other role.

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
   - secure: ij5x06dAGkYapI8iv+Ii5nbsrLw5YhAfQ6yhIuFFsEu8+Uh96wAGOSNBqRcUMhSpCtgS3pCMhcDUU2BEYHUkjsnYGa+8NAMsWLGTJaiPihxjGpeNW8tfJsfXVotYGVq5tI2fBkUqBuj3OhTCLIbE5KPDRJPpCJaowg9EJwboB++yX+pNVrgeI758C4hV+bSEiNlUpnVv60gbZx3BY/SJWWlAit7QtKdEAMviQf94WfsHYIws+NOc8566M22VURh+dkR//K/y1w7jBrHnGlt2P7ZTeYLriSAkPlpxscVuZYSUM+qkcma5L3VqPjij+C2UU1o0SbwlUEN4kwEKj434vC0E6iAC94gzGwJ+zd/K5m7Qs/bbKExNFC19T8cKWW+MizPNYqwu48svhy0mKGNR7P1AvH2c1ndcbdHbk3lIomA4FTNPhzD23FU+00XrR5JGrhkuTrXOLvXaxWx1mJx8RD3KvSFNfau9lnGbyYixvT4KhMs1/F/rvUrLE0KifX5u6QAUQWaPTMp4HT7xo0/8ahCixgR3T1PzlvQffyUkyL19NHX6uy5ih5MqN6sm43BMcPWLh2N6rcEZKs2UgNTieJFky+nLwiHy/ZBRCdvUc/NoOWdT2ubgNrYTe9fXh3JfSN+EtPHB77P6TELEq05zMTpWTZTLVxsSSf4GMYmrMGQ=
```
Comment (hugh-emerson) [inline .travis.yml:4]: Are a lot of these secrets now safe to be removed? If they are for Trello settings etc.

File: app.py [hunk 3]
```diff
@@ -1,42 +1,110 @@
...
 
+secret_key = os.getenv("SECRET_KEY")
 mongo_instance = os.getenv("MONGO_INSTANCE")
 mongo_user = os.getenv("MONGO_USER")
 mongo_password = os.getenv("MONGO_PASSWORD")
+oauth_client_id = os.getenv("OAUTH_CLIENT_ID")
+oauth_client_secret = os.getenv("OAUTH_CLIENT_SECRET")
+
+
 srv = "+srv" if mongo_instance else ""
```
Comment (hugh-emerson) [inline app.py:24]: If `mongo_instance` isn't set, doesn't everything break?

File: .travis.yml (hunk 2 above)
Comment (waormd) [reply .travis.yml:4]: No, already cleaned these up in previous modules

File: app.py (hunk 3 above)
Comment (waormd) [reply app.py:24]: This is for the local stub mongo that doesnt use +srv. I appreciate that prod code shouldnt be written for test but I spent a while trying to get this working at the time and the +srv that is required to connect to the remote instance doesnt seem to play ball with the local stub

File: README.md [hunk 4]
```diff
@@ -1,30 +1,23 @@
...
-
-#### Run Dev (cmd.exe)
-```cmd
-docker build --target dev --tag waormd/todo-app-dev . && \
-docker run -p 5000:5000/tcp -d -v %cd%:/todo-app --env-file .env waormd/todo-app-dev
+Create a .env file which should contain the following environment variables:
+```
+MONGO_INSTANCE  = <instance>
+MONGO_USER      = <user>
+MONGO_PASSWORD  = <password>
```
[the remark this answers is not in the document]
Comment (waormd) [reply README.md]: #10 c8fe8f6d1f98cb1d8ee577861dd50d9e73fbfab5

File: app.py (hunk 1 above)
Comment (waormd) [reply app.py:39]: Good point

File: app.py [hunk 5]
```diff
@@ -1,42 +1,110 @@
...
+login_manager.init_app(app)
+
+@login_manager.unauthorized_handler
+def unauthenticated():        
+    uri = oauth_client.prepare_request_uri('https://github.com/login/oauth/authorize')
+    return redirect(uri) 
+
+@login_manager.user_loader    
+def load_user(user_id):        
+    return User(user_id)
```
[the remark this answers is not in the document]
Comment (waormd) [reply app.py:103]: Good point

[the remark this answers is not in the document]
Comment (waormd) [reply app.py]: 46d8ebad49485e005cf6b22f7c5f14e6d9a975ad

**Pull request merged**